content/kubermatic/v2.22/installation/local-kkp-installation/_index.en.md

```diff
@@ -0,0 +1,83 @@
+
+1. Download the installer . This is the only manual step; there's no need to prepare any configuration since the installer should automatically configure KKP.
+
+```bash
+VERSION=$(curl -w '%{url_effective}' -I -L -s -S https://github.com/kubermatic/kubermatic/releases/latest -o /dev/null | sed -e 's|.*/v||')
+wget https://github.com/kubermatic/kubermatic/releases/download/v${VERSION}/kubermatic-ce-v${VERSION}-linux-amd64.tar.gz
+tar -xzvf kubermatic-ce-v${VERSION}-linux-amd64.tar.gz
+```
+You can find more information regarding the download instructions in the [CE installation guide](../install-kkp-ce/#download-the-installer).
+
+2. Run the `local` command:
+
+```bash
+./kubermatic-installer local kind
+```
+
+The KKP installation should take no longer than a couple of minutes.
+First, it creates a `kind` cluster named `kind-kkp-cluster`, exposes ports `443`, `80` for the KKP API, KKP dashboard, dex, and ports `6443`, and `8088` for user cluster `kube-apiserver` via the [tunneling expose strategy](../../tutorials-howtos/networking/expose-strategies/#tunneling). You can examine the generated `kind` config, as well as the generated Kubermatic and Helm values configuration, under your working directory:
+
+```none
+./examples/kind-config.yaml
+./examples/kubermatic.yaml
+./examples/values.yaml
+```
+
+The installer also configures a single `Seed` and `Presets` for KubeVirt user clusters deployed in the same `kind` cluster. You can inspect the resources using `kubectl`:
+
+```bash
+kubectl get seed -nkubermatic kubermatic -o yaml 
+kubectl get preset -nkubermatic local -o yaml
+```
+
+## Post-Installation
+
+After completing the installation, you should have a local, trimmed-down KKP setup suitable for development purposes. The installer will provide login instructions:
+
+```none
+INFO[0408] KKP installed successfully, login at http://10.0.0.12.nip.io
+INFO[0408]   Default login:    kubermatic@example.com
+INFO[0408]   Default password: password
+```
+
+## Teardown Procedure
+
+When you're finished experimenting with your local KKP setup, you can easily terminate it with a single command:
+
+```bash
+kind delete cluster --name kkp-cluster
+```
+
+## Configuration and Customization
+
+* The KKP dashboard is [exposed using `nip.io`](https://nip.io/), and certain browser plugins, such as various ad blockers, may prevent access to `nip.io`. If this is the case, consider disabling these plugins.
+* By default, KubeVirt is configured to use hardware virtualization. If this is not possible for your setup, consider [setting KubeVirt to use software emulation mode](https://github.com/kubevirt/kubevirt/blob/v1.0.0-rc.0/docs/software-emulation.md).
+* On Linux, KubeVirt uses the inode notify kernel subsystem `inotify` to watch for changes in certain files. You may need to set the default values higher to ensure KubeVirt operates correctly.
+```bash
+sudo sysctl -w fs.inotify.max_user_watches=524288
+sudo sysctl -w fs.inotify.max_user_instances=256
```

Comment:
@xmudrii I used the values you provided, I tried to search KubeVirt docs for any recommendations but I couldn't find any.

Maybe @mfranczy has some experience with this?

Reply:
Those values are very biased and dependent on processes that watch for files on your system and if your hardware is very memory constrained (like 4Gi) it might be a bad idea to go that high.

Usually you don't need to fine-tune this just for KubeVirt, especially when you start a development setup which creates a few VMs. Also as I remember inotify was only used to trigger gracefule shutdown of VMIs and it's a legacy support https://github.com/kubevirt/kubevirt/blob/292f07f9833011246f9aa9d6780eba094d5cfd0a/cmd/virt-handler/virt-handler.go#L303-L309

I am pretty sure that KubeVirt doesn't need to go with such high values. Rather, if user has a problem I would advise that they judge how much they need to increase.

@wozniakjan why do you advise to change this? Do you have evidence that this is really required for KubeVirt? Maybe you discovered some intofy leak.

Reply:
If you want to come with real numbers for KubeVirt I would suggest to run dev setup and check how many inotify watches are open. You can check this with `lsof`  and filter `inotify` over name column.

Reply:
> why do you advise to change this? 

This was discovered by @xmudrii, the `kubevirt/virt-handler` was crashing for him with `"Failed to create an inotify watcher"`, more here - https://github.com/kubermatic/kubermatic/issues/12340#issuecomment-1602433880

But the verbiage is purposefully crafted as a "you may need .. if KubeVirt is failing". You sense correctly that I would prefer not to direct users or even try to manage this inside of the installer.

> You can check this with `lsof` and filter `inotify` over name column.

This is really handy, thanks!

Reply:
I added additional sentence supporting ^^
```
Reasonably elevated values are described below but it's recommended to inspect your system to figure out the correct `inotify` values depending on your needs and current setup
```

Reply:
@mfranczy are you happy with the changes done here?